### DISCUSSION TITLE
HTTP API iobroker or another sources

### DISCUSSION THREAD

**Discussion by amnesia106** [fabian-lauer/dbus-shelly-3em-smartmeter#43] [Q&A]
Hi, and great work! that saves a lot of users the em24 enery meter.

it is possible to integrate every data over http api in JSON format? without high costs in my meter housing i cant integrate a em24. a shelly e3m is possible, but i have all energy data on iobroker (generation, grid relation and grid feeding). iobroker can provide with a plugin all data over http GET in JSON format. all values of my energy meters phases shown in total. 

https://github.com/fabian-lauer/dbus-shelly-3em-smartmeter/discussions/5

Hi, erstmal top Arbeit, das erspart vielen den em24.

Mich würde interessieren ob es möglich ist, auch auf Daten von iobroker (http api) zuzugreifen? Ohne größere Umbauarbeiten am Zählerschrank (ist voll bei mir) bekomme ich kein em24 rein. Jedoch bekomme ich die Werte von Bezug, Einspeisung und Erzeugung von meinem Wechselrichter und smartmeter direkt. Diese könnten per iobroker simpleapi per http JSON zu Verfügung gestellt werden. Abtastraten ab 1 Sekunde sind dort möglich.

https://github.com/ioBroker/ioBroker.simple-api

sieht dann z.b. folgendermaßen aus:

```
http://192.168.2.8:8090/getPlainValue/javascript.0.Haus_Strom_Verbrauch_gesamt?json 

1158
```
Es wird der Bezug, Einspeisung und Eigenverbrauch saldierend anzeigt.

Vielleicht kann ich unterstützend mitwirken oder bei der Machbarkeit helfen…

**Comment by fabian-lauer** [comment]
Hi,

I would say that this is doable, but I have still some logical error in the script - but this is based on the Shelly 3EM logic of blanacing/sum.

I guess what you want to do is very easy in Python - you just need to call your endpoint for "every" value - may take longer, but should be ok.

I am not familiar with iobroker yet - is this structure a common one or specific for your setup?
Asking, because it would make sense to implement in in a way, more people can use it....

**Comment by amnesia106** [comment]
with iobroker and the simple api adapter everybody can provide any values. with iobroker you can change the variables like you want. example: when this script needs a "grid" ; "generation" or something else, every user can rename or dublicate his variable which is needed for the script. I think the most iobroker users have three values. energy generation, grid relation and feeding.

exampe i open a variable in a browser

`http://192.168.2.8:8090/getPlainValue/javascript.0.Haus_Strom_Bezug_Netz?json `

response is only the value of this variable 

`0`

for each variable for example

```
http://192.168.2.8:8090/getPlainValue/javascript.0.Piko.Ausgangsleistung?json

4456,4447
```

```
http://192.168.2.8:8090/getPlainValue/javascript.0.Einspeisung?json

2541
```

the structure is common - `http://ipiobrokerserver:portfromsimpleapi/getPlainValue/yourvariable?json`
this gives the raw output from that value without any extras

all values are in watt in my case

**Comment by fabian-lauer** [comment]
Closing this issue as it is more a Q&A regarding ioBroker - will move to discussion section

**Comment by LecramRed** [comment]
This would be great. I have a P1 meter and get this data:
`{
  "wifi_ssid": "DMSweg_RU",
  "wifi_strength": 52,
  "smr_version": 50,
  "meter_model": "Kaifa AIFA-METER",
  "unique_id": "xxxxxxxxxxxxxxx",
  "active_tariff": 2,
  "total_power_import_kwh": 5394.007,
  "total_power_import_t1_kwh": 2851.56,
  "total_power_import_t2_kwh": 2542.447,
  "total_power_export_kwh": 457.802,
  "total_power_export_t1_kwh": 189.506,
  "total_power_export_t2_kwh": 268.296,
  "active_power_w": -7714,
  "active_power_l1_w": -3255,
  "active_power_l2_w": -3024,
  "active_power_l3_w": -1433,
  "active_voltage_l1_v": 234.5,
  "active_voltage_l2_v": 233.8,
  "active_voltage_l3_v": 234.4,
  "active_current_l1_a": -13.881,
  "active_current_l2_a": -12.934,
  "active_current_l3_a": -6.113,
  "voltage_sag_l1_count": 2,
  "voltage_sag_l2_count": 0,
  "voltage_sag_l3_count": 2,
  "voltage_swell_l1_count": 0,
  "voltage_swell_l2_count": 0,
  "voltage_swell_l3_count": 1,
  "any_power_fail_count": 15,
  "long_power_fail_count": 7,
  "external": [
    
  ]
}`
